### PR TITLE
Noam/dee 152 add quickstart under a new nlp section

### PR DESCRIPTION
**Pull request by noamzbr**
Add documentation to the nlp package + new quickstart

Still missing:
- [x] Fix the checks API section - waiting for help by @ItayGabbay. Until fix links in new quickstart are broken
- [x] Add a mention of it in quickstart. Can be added once https://github.com/deepchecks/deepchecks/pull/2372 is merged to avoid conflicts.

### CODE REVIEW

File: deepchecks/nlp/text_data.py [hunk 1]
```diff
@@ -347,6 +348,17 @@ def calculate_default_properties(self, include_properties: t.List[str] = None,
...
+    def set_properties(self, properties: pd.DataFrame):
+        """Set the properties of the dataset."""
+        if self._properties is not None:
+            warnings.warn('Properties already exist, overwriting them', UserWarning)
+
+        if not isinstance(properties, pd.DataFrame):
+            raise DeepchecksValueError('properties must be a pandas DataFrame')
+        if list(properties.index) != self.index:
+            raise DeepchecksValueError('properties index must be the same as the dataset index')
+        self._properties = properties
```
Comment (nirhutnik) [inline deepchecks/nlp/text_data.py:360]: 1. Add set_metadata as well. 
2. **remove** the parallel code in the __init__ and just use this function in the __init__.
3. In context.assert_properties/metadata, tell the user they can use "set_x" functions to add these (instead / in addition to initialize the object)

File: docs/source/user-guide/nlp/quickstarts/README.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
+NLP Quickstarts
+===============
+
+Deepchecks for NLP is currently in alpha stage. The following quickstart will walk you through how you can try the
+existing functionality quickly for your own NLP project.
+
+In addition to the individual quickstarts available here, you can also use built-in
+:ref:`Suites <deepchecks_hierarchy_suite>` to run all the checks relevant to a specific part in your ML pipeline.
+For the API reference of each of the built-in Suites, check out :mod:`deepchecks.nlp.suites`.
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/README.rst:9]: How about adding a link to our community slack and inviting users to give us feedback?

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+"""
+Quickstart - Getting Started
+****************************
+
+In order to run deepchecks for nlp all you need to have are the following for both your train and test data:
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:6]:
```suggestion
In order to run deepchecks for NLP all you need to have are the following for both your train and test data:
```

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+1. Your text data - a list of strings, each string is a single sample (can be a sentence, paragraph, document etc.)
+2. Your labels - either a Text Classification label or a Token Classification label
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:9]: Capitalization of these words is weird.
You can keep it if they were the links to the relevant docs - you can link each one to the relevant part in the guide

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+   (see :doc:`Supported Tasks </user-guide/nlp/supported_tasks>` for more info)
+3. Your models predictions (see :doc:`Supported Tasks </user-guide/nlp/supported_tasks>` for info on supported formats
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:11]:
```suggestion
3. Your models predictions (see :doc:`Supported Tasks </user-guide/nlp/supported_tasks>` for info on supported formats).
```

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 6]
```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+"""
+Quickstart - Getting Started
+****************************
+
+In order to run deepchecks for nlp all you need to have are the following for both your train and test data:
+
+1. Your text data - a list of strings, each string is a single sample (can be a sentence, paragraph, document etc.)
+2. Your labels - either a Text Classification label or a Token Classification label
+   (see :doc:`Supported Tasks </user-guide/nlp/supported_tasks>` for more info)
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:10]:
```suggestion
   (see :doc:`Supported Tasks </user-guide/nlp/supported_tasks>` for more info).
```

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 7]
```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+"""
+Quickstart - Getting Started
+****************************
+
+In order to run deepchecks for nlp all you need to have are the following for both your train and test data:
+
+1. Your text data - a list of strings, each string is a single sample (can be a sentence, paragraph, document etc.)
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:8]:
```suggestion
1. Your text data - a list of strings, each string is a single sample (can be a sentence, paragraph, document etc.).
```

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+2. Your labels - either a Text Classification label or a Token Classification label
+   (see :doc:`Supported Tasks </user-guide/nlp/supported_tasks>` for more info)
+3. Your models predictions (see :doc:`Supported Tasks </user-guide/nlp/supported_tasks>` for info on supported formats
+
+If you don't have deepchecks installed yet:
+
+.. code:: python
+
+    # If you don't have deepchecks installed yet:
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:17]: Repeating "if you don't" twice

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+train.head()
+
+#%%
+#
+# We can see that we have the tweet text itself, the label (the emotion) and then some additional metadata columns.
+#
+# We can now create a :class:`TextData <deepchecks.nlp.TextData>` object for each of the train and test dataframes.
+# This object is used to pass your data to the deepchecks checks.
+#
+# To create a TextData object, the minimum required arguments is the text itself, but passing only the text
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:58]:
```suggestion
# To create a TextData object, the only required argument is the text itself, but passing only the text
```

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+# will prevent multiple checks from running. In this example we'll pass the label as well and also provide
+# Metadata (the other columns in the dataframe) which we'll use later on in the guide. Finally, we'll also
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:60]:
```suggestion
# metadata (the other columns in the dataframe) which we'll use later on in the guide. Finally, we'll also
```

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+                 index=train.index, metadata=train.drop(columns=['label', 'text']))
+test = TextData(test.text, label=test['label'], task_type='text_classification',
+                index=test.index, metadata=test.drop(columns=['label', 'text']))
+
+#%%
+# Building a Model
+# ================
+#
+# In this example we'll train a very simple model using the CatBoostClassifier, trained over the embeddings of the
+# tweets. These embeddings where created using the OpenAI GPT-3 model. You can calculate your own embeddings using
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:78]:
```suggestion
# tweets. These embeddings were created using the OpenAI GPT-3 model. You can calculate your own embeddings using
```

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 12]
```diff
@@ -0,0 +1,213 @@
...
+#
+
+train = TextData(train.text, label=train['label'], task_type='text_classification',
+                 index=train.index, metadata=train.drop(columns=['label', 'text']))
+test = TextData(test.text, label=test['label'], task_type='text_classification',
+                index=test.index, metadata=test.drop(columns=['label', 'text']))
+
+#%%
+# Building a Model
+# ================
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:75]: What's missing for me:
1. Saying that we know this is weird, and we're using a classic model just for simplicity
2. Explain that the best practice is that the user use their own model's embeddings
3. In addition to 2, say explicitly they don't have to use our embeddings calculator.

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 13]
```diff
@@ -0,0 +1,213 @@
...
+#%%
+# Running Deepchecks
+# ==================
+#
+# Now that we have our data and model, we can run our first checks!
+#
+# Performance Checks
+# ------------------
+#
+# We'll start by running the
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:105]: Consider by starting with the label drift

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 14]
```diff
@@ -0,0 +1,213 @@
...
+
+#%%
+# As we can see, the check found a segment that has significantly lower performance than the rest of the data. In the
+# first tab of the display we can see that there is a large segment of young Europeans that have significantly lower
+# performance than the rest of the data. Perhaps there is some language gap here?
+#
+# Properties
+# ^^^^^^^^^^
+#
+# Properties are one-dimension values that are extracted from the text. Among their uses they can be used to
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:152]:
```suggestion
# Properties are one-dimension values that are extracted from the text. Among their uses, they can be used to
```

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 15]
```diff
@@ -0,0 +1,213 @@
...
+#    DataFrame of properties to the TextData `properties` argument, or by calling the
+#    :meth:`set_properties <deepchecks.nlp.TextData.set_properties>` method anytime later with such a DataFrame. You
+#
+# In this example we'll use already calculated properties, which we'll load from a file to save time, but we recommend
+# that for your own ``TextData`` objects you calculate the properties automatically using the first method.
+
+# Set the properties
+train_properties, test_properties = tweet_emotion.load_properties(as_train_test=True)
+train.set_properties(train_properties)
+test.set_properties(test_properties)
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:174]: Use calculate_properties instead

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 16]
```diff
@@ -0,0 +1,213 @@
...
+
+#%%
+# As we can see, the check found a segment that has significantly lower performance than the rest of the data on our
+# computed properties. For example, in the second tab of the display we can see that the model has significantly lower
+# performance on tweets with positive sentiment.
+#
+# Data Integrity Checks
+# ---------------------
+#
+# These previous checks were all about the model's performance. Now we'll run a check that attmepts to find instances
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:191]:
```suggestion
# These previous checks were all about the model's performance. Now we'll run a check that attempts to find instances
```

File: docs/source/user-guide/nlp/supported_tasks.rst [hunk 17]
```diff
@@ -96,27 +97,38 @@ Multilabel Predictions
...
 
+For token classification tasks labels and predictions are given in any
+`IOB format <https://en.wikipedia.org/wiki/Inside%E2%80%93outside%E2%80%93beginning_(tagging)>`__
+supported by the `seqeval <https://github.com/chakki-works/seqeval>`__ library. The label should be passed as a
+|sequence| of sequences, with the inner |sequence| containing the appropriate IOB annotation for each token in the sample.
+
+To let deepchecks know what are the individual tokens in the text sample, it's **highly recommended** that you pass a
+list of the tokens to the ``tokenized_text`` argument of the :class:`TextData <deepchecks.nlp.TextData>`
+constructor method. Otherwise, deepchecks will attempt to tokenize the text samples (given to the ``text`` argument)
+by splitting them on spaces.
```
Comment (nirhutnik) [inline docs/source/user-guide/nlp/supported_tasks.rst:108]:
```suggestion
by splitting them by spaces.
```

**Review by nirhutnik — APPROVED** [review]

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py [hunk 18]
```diff
@@ -0,0 +1,213 @@
...
+result.show()
+
+#%%
+# In this case the check didn't find any properties that have a high correlation with the labels. Apart from the
+# sentiment property, which is expected to have high relevance to the emotion of the tweet, the other properties
+# have very low correlation to the label.
+#
+# You can find the full list of available NLP checks in the :mod:`api documentation <deepchecks.nlp.checks>`.
+
+# sphinx_gallery_thumbnail_path = '_static/images/sphinx_thumbnails/nlp_quickstarts/getting_started.png'
```
[the remark this answers is not in the document]
Comment (noamzbr) [reply docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:213]: No need for new line here

File: docs/source/user-guide/nlp/quickstarts/plot_getting_started.py (hunk 13 above)
Comment (noamzbr) [reply docs/source/user-guide/nlp/quickstarts/plot_getting_started.py:105]: why not prediction drift? More interesting

**Comment by noamzbr** [comment]
Replaces https://github.com/deepchecks/deepchecks/pull/2383

**Pull request closed without merging**